### ISSUE TITLE
TypeError: util.isDate is not a function

### ISSUE THREAD

**Issue by dabrain34** [LinusU/node-appdmg#243]
Since today, I'm unable to use appdmg anymore getting the error:

```
make macos installer(.dmg)......
[ 1/21] Looking for target...                [ OK ]
[ 2/21] Reading JSON Specification...        [ OK ]
[ 3/21] Parsing JSON Specification...        [ OK ]
[ 4/21] Validating JSON Specification...     [ OK ]
[ 5/21] Looking for files...                 [ OK ]
[ 6/21] Calculating size of image...         [ OK ]
[ 7/21] Creating temporary image...          [ OK ]
[ 8/21] Mounting temporary image...          [ OK ]
[ 9/21] Making hidden background folder...   [ OK ]
[10/21] Copying background...                [ OK ]
[11/21] Reading background dimensions...     [ OK ]
[12/21] Copying icon...                      [ OK ]
[13/21] Setting icon...                      [ OK ]
[14/21] Creating links...                    [ OK ]
[15/21] Copying files...                     [ OK ]
[16/21] Making all the visuals...            
TypeError: util.isDate is not a function
[failed]
make macos portable......[done]
```
See https://gitlab.freedesktop.org/dabrain34/GstPipelineStudio/-/jobs/66330552

A few weeks ago it was complaining about 

```
[16/21] Making all the visuals...            (node:60904) [DEP0047] DeprecationWarning: The `util.isDate` API is deprecated.  Please use `arg instanceof Date` instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
```
https://gitlab.freedesktop.org/dabrain34/GstPipelineStudio/-/jobs/64591511/raw

**Comment by dabrain34** [comment]
I checked the version of node and I was using v22.9.0 instead of the new one which is v23.1.0

**Comment by scosman** [comment]
beat me to it by 7 mins. Same issue on node v23.1

**Comment by dabrain34** [comment]
I tried to install v22.9.0 with

nvm install v22.9.0

But appdmg is not correctly installed then.

**Comment by scosman** [comment]
This seems to be the source: https://github.com/LinusU/node-alias/blob/fb006c2ad35af646193e8744d7b43ef95c495af6/lib/encode.js#L8

**Comment by dabrain34** [comment]
good pick !

**Comment by dabrain34** [comment]
https://github.com/LinusU/node-alias/pull/20

**Comment by bahricanyesil** [comment]
How can we temporarily fix this @dabrain34 @scosman ?

**Comment by scosman** [comment]
Just use node 20 for now. I use nvm to switch it.

**Comment by bahricanyesil** [comment]
Thank you. I will use it temporarily until the related PR is merged.

> Just use node 20 for now. I use nvm to switch it.

**Comment by dabrain34** [comment]
https://gitlab.freedesktop.org/dabrain34/GstPipelineStudio/-/merge_requests/69/diffs

**Comment by sinosoidal** [comment]
@pzhlkj6612 this fix works. Is this going to be merged? @LinusU are you still maintaining this project?

**Comment by LinusU** [comment]
Merged and released in node-alias: [🚢 0.2.12 / 2024-11-20](https://github.com/LinusU/node-alias/releases/tag/v0.2.12)
